### PR TITLE
Add SingleBitOperations

### PR DESCRIPTION
**Pull request by lukasb1b**
#4350 as its not done yet

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

File: src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.thealgorithms.bitmanipulation;
+
+/*
+ * Author: lukasb1b (https://github.com/lukasb1b)
+ */
+
+public class SingleBitOperators {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java:7]: Let's make this class a _proper utility class_.

```suggestion
public final class SingleBitOperators {
    private SingleBitOperators() {
    }
```

File: src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+/*
+ * Author: lukasb1b (https://github.com/lukasb1b)
+ */
+
+public class SingleBitOperators {
+    /**
+     * Flip the bit at position 'bit' in 'num'
+     */
+    public static int flipBit(int num, int bit) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java:11]:
```suggestion
    public static int flipBit(final int num, final int bit) {

```

File: src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        return num ^ (1 << bit);
+    }
+    /**
+     * Set the bit at position 'bit' to 1 in the 'num' variable
+     */
+    public static int setBit(int num, int bit) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java:17]:
```suggestion
    public static int setBit(final int num, final int bit) {

```

File: src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        return num | (1 << bit);
+    }
+    /**
+     * Clears the bit located at 'bit' from 'num'
+     */
+    public static int clearBit(int num, int bit) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java:23]:
```suggestion
    public static int clearBit(final int num, final int bit) {

```

File: src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        return num & ~(1 << bit);
+    }
+    /**
+     * Get the bit located at 'bit' from 'num'
+     */
+    public static int getBit(int num, int bit) {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperators.java:29]:
```suggestion
    public static int getBit(final int num, final int bit) {

```

File: src/test/java/com/thealgorithms/bitmanipulation/SingleBitOperatorTest.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
+package com.thealgorithms.bitmanipulation;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (vil02) [inline src/test/java/com/thealgorithms/bitmanipulation/SingleBitOperatorTest.java:3]:
```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;

```

**Review by vil02 — CHANGES_REQUESTED** [review]
I would suggest to rename this class to `SingleBitOperations`.

File: src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperator.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
+package com.thealgorithms.bitmanipulation;
+
+/*
+ * Author: lukasb1b (https://github.com/lukasb1b)
+ */
+
+public final class SingleBitOperator {
```
Comment (vil02) [inline src/main/java/com/thealgorithms/bitmanipulation/SingleBitOperator.java:7]: As suggested [earlier](https://github.com/TheAlgorithms/Java/pull/4415#discussion_r1338339851):
```suggestion
public final class SingleBitOperator {
    private SingleBitOperator() {
    }
```

**Review by vil02 — CHANGES_REQUESTED** [review]
Why do you use the name `SingleBitOperator` and not `SingleBitOperations`?

I think the second name makes more sense in this context:
- this class defines _single bit operations_,
- it is not a _single bit operator_.

Please rebase (cf. #4422).

**Review by lukasb1b — COMMENTED** [review]
Why does the programm need this?

**Comment by vil02** [comment]
> Why does the programm need this private constructor?

Your class has only `static` methods. Therefore it does not really make sense to have objects of this class. Marking the default constructor as `private` (and not having other ones) makes it impossible to create objects of that class. Without this, one could still write something like `var obj = SingleBitOperator();`, but as said earlier, it does not really make sense.

Also JaCoCo will show now _100%_ coverage for your file.

**Comment by lukasb1b** [comment]
Ok that does make sense

**Comment by vil02** [comment]
@lukasb1b please have a look at [this comment](https://github.com/TheAlgorithms/Java/pull/4415#pullrequestreview-1649293416) about renaming.

**Comment by lukasb1b** [comment]
@vil02 changed the naming

**Review by vil02 — APPROVED** [review]

**Pull request merged**